numpy/core/setup_common.py

```diff
@@ -102,12 +102,14 @@ def set_sig(sig):
     args = args.rpartition(")")[0]
     funcname = prefix.rpartition(" ")[-1]
     args = [arg.strip() for arg in args.split(",")]
-    FUNC_CALL_ARGS[funcname] = ", ".join("(%s) 0" % arg for arg in args)
+    # We use {0} because 0 alone cannot be cast to complex on MSVC in C:
+    FUNC_CALL_ARGS[funcname] = ", ".join("(%s){0}" % arg for arg in args)
```

Comment:
Lets see if this works.  I had to add the struct here because `(_Dcomplex)0` is not a valid cast in MSVC `complex.h`...